tests/CommonExtractor.php

```diff
@@ -0,0 +1,497 @@
+                    ON c.TABLE_NAME = kcu.TABLE_NAME AND c.COLUMN_NAME = kcu.COLUMN_NAME";
+        } else {
+            $sql = "SELECT c.*
+                    FROM INFORMATION_SCHEMA.COLUMNS as c";
+        }
+
+        $sql .= $whereClause;
+
+        $sql .= " ORDER BY c.TABLE_SCHEMA, c.TABLE_NAME, ORDINAL_POSITION";
+
+        /** @var \PDOStatement $res */
+        $res = $this->getConnection()->query($sql);
+        /** @var array $rows */
+        $rows = $res->fetchAll(\PDO::FETCH_ASSOC);
+
+        foreach ($rows as $i => $column) {
+            $curTable = $column['TABLE_SCHEMA'] . '.' . $column['TABLE_NAME'];
+            $length = ($column['CHARACTER_MAXIMUM_LENGTH']) ? $column['CHARACTER_MAXIMUM_LENGTH'] : null;
+            if (is_null($length) && !is_null($column['NUMERIC_PRECISION'])) {
+                if ($column['NUMERIC_SCALE'] > 0) {
+                    $length = $column['NUMERIC_PRECISION'] . "," . $column['NUMERIC_SCALE'];
+                } else {
+                    $length = $column['NUMERIC_PRECISION'];
+                }
+            }
+            $curColumn = [
+                "name" => $column['COLUMN_NAME'],
+                "sanitizedName" => \Keboola\Utils\sanitizeColumnName($column['COLUMN_NAME']),
+                "type" => $column['DATA_TYPE'],
+                "primaryKey" => ($column['COLUMN_KEY'] === "PRI") ? true : false,
+                "length" => $length,
+                "nullable" => ($column['IS_NULLABLE'] === "NO") ? false : true,
+                "default" => $column['COLUMN_DEFAULT'],
+                "ordinalPosition" => $column['ORDINAL_POSITION'],
+            ];
+
+            if (array_key_exists('CONSTRAINT_NAME', $column) && !is_null($column['CONSTRAINT_NAME'])) {
+                $curColumn['constraintName'] = $column['CONSTRAINT_NAME'];
+            }
+            if (array_key_exists('REFERENCED_TABLE_NAME', $column) && !is_null($column['REFERENCED_TABLE_NAME'])) {
+                $curColumn['foreignKeyRefSchema'] = $column['REFERENCED_TABLE_SCHEMA'];
+                $curColumn['foreignKeyRefTable'] = $column['REFERENCED_TABLE_NAME'];
+                $curColumn['foreignKeyRefColumn'] = $column['REFERENCED_COLUMN_NAME'];
+            }
+            if ($column['EXTRA']) {
+                $curColumn["extra"] = $column["EXTRA"];
+                if ($column['EXTRA'] === 'auto_increment') {
+                    $curColumn['autoIncrement'] = $tableDefs[$curTable]['autoIncrement'];
+                }
+                if ($column['EXTRA'] === 'on update CURRENT_TIMESTAMP'
+                    && $column['COLUMN_DEFAULT'] === 'CURRENT_TIMESTAMP') {
+                    $tableDefs[$curTable]['timestampUpdateColumn'] = $column['COLUMN_NAME'];
+                }
+            }
+            $tableDefs[$curTable]['columns'][$column['ORDINAL_POSITION'] - 1] = $curColumn;
+        }
+        return array_values($tableDefs);
+    }
+
+    public function testConnection(): void
```

Comment:
Aha, to nebylo vůbec obvious, že má testConnection házet UserException. Jak jsem to měl zjistit?

Reply:
nijak, testConnection muze hazet co chce, tady ten try-catch je tam uplne zbytecne